pkg/tracing/ebpf/core_structures.h

```diff
@@ -0,0 +1,120 @@
+#ifndef __CORE_STRUCTURES_H__
+#define __CORE_STRUCTURES_H__
+
+#include <vmlinux.h>
+
+/*
+ * All structs and unions in this file should have a "preserve access index"
+ * attribute. The following attaches this attribute to all records (structs,
+ * unions, classes).
+ * @see https://clang.llvm.org/docs/LanguageExtensions.html
+ */
+#pragma clang attribute push
+#pragma clang attribute(__attribute__((preserve_access_index)),                \
+                        apply_to = record)
+
+// this is not core structure per se, but it would have been defined in a full
+// vmlinux.h
+enum {
+  false = 0,
+  true = 1,
+};
+
+enum {
+  TCP_ESTABLISHED = 1,
+  TCP_SYN_SENT = 2,
+  TCP_SYN_RECV = 3,
+  TCP_FIN_WAIT1 = 4,
+  TCP_FIN_WAIT2 = 5,
+  TCP_TIME_WAIT = 6,
+  TCP_CLOSE = 7,
+  TCP_CLOSE_WAIT = 8,
+  TCP_LAST_ACK = 9,
+  TCP_LISTEN = 10,
+  TCP_CLOSING = 11,
+  TCP_NEW_SYN_RECV = 12,
+  TCP_MAX_STATES = 13,
+};
+
+struct user_pt_regs {
```

Comment:
This is not a core structure - please move to an "arm_support.h" file

Reply:
done